lib/miq_expression/field.rb

```diff
@@ -82,6 +82,19 @@ def contains(other)
     )
   end
 
+  def custom_attribute_column_name
+    column.gsub(CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX, "")
+  end
+
+  def column_type
+    if custom_attribute_column?
+      field_type = CustomAttribute.find_by(:name => custom_attribute_column_name).try(:serialized_value).class
+      field_type == NilClass ? :string : field_type.to_s.downcase.to_sym
```

Comment:
You don't need to do `==` here, you can fall back on the truthy/falsey value of `field_type`:
```ruby
field_type ? field_type.to_s.downcase.to_sym : :string
```

Also, I'm concerned that this is reaching deeply into `CustomAttribute`'s data. Could this be moved to that class instead, and tests added? It's not clear to me from this code what range of values I might expect. Are they all valid column types?